test/org_parser/parser_test.cljc

```diff
@@ -1004,3 +1029,60 @@ is another section"))))))
                [:table-formula "$4=vmean($2..$3)"]]]
              (parse " |--+--|\n | x|x |\n |--+--|\n #+TBLFM: $4=vmean($2..$3)"))))
     ))
+
+
+(deftest clock
+  (let [parse #(parser/org % :start :clock)]
+    (testing "a simple clock line"
+      (is (= [:clock [:timestamp-inactive-range
+               [:ts-inner-w-time [:ts-date "2021-05-22"] [:ts-day "Sat"] [:ts-time "23:26"]]
+               [:ts-inner-w-time [:ts-date "2021-05-22"] [:ts-day "Sat"] [:ts-time "23:46"]]]
+              [:clock-duration [:clock-dur-hh "0"] [:clock-dur-mm "20"]]]
+             (parse "CLOCK: [2021-05-22 Sat 23:26]--[2021-05-22 Sat 23:46] =>  0:20"))))
+    (testing "a simple clock line"
+      (is (= [:clock [:timestamp-inactive-range [:ts-inner-span
+               [:ts-inner-w-time [:ts-date "2021-05-22"] [:ts-day "Sat"] [:ts-time "23:26"]]
+               [:ts-time "23:46"]
+               [:ts-modifiers]]]
+              [:clock-duration [:clock-dur-hh "0"] [:clock-dur-mm "20"]]]
+             (parse " CLOCK: [2021-05-22 Sat 23:26-23:46] =>  0:20 "))))
```

Comment:
@schoettl It's pretty cool that the parser can read the shorthand. Can Emacs org mode, though? It doesn't seem to work for me and I also haven't found documentation for Org mode that shows this syntax for `CLOCK` entries.

I mean, if the parser can do it 'by accident', then that's totally fine, of course. However, I'm curious if it's part of the standard and where that would be documented.

Reply:
The worg spec says, there must be an inactive timestamp range after `CLOCK: `. Timestamps are defined in the spec as:
```
...
[DATE TIME REPEATER-OR-DELAY]--[DATE TIME REPEATER-OR-DELAY]   (inactive range)
[DATE TIME-TIME REPEATER-OR-DELAY]                             (inactive range)
```

So I think I stick to the worg spec. But what is the real spec? Org mode or the [worg spec](https://orgmode.org/worg/dev/org-syntax.html#Timestamp)? ;) Our org-parser README says, the worg spec leads to the reference implementation. It might also be vice versa or so so ^^

Reply:
In any case, I think it's good to allow all possible timestamp ranges here.

Reply:
Well, the worg 'spec' is a very traditional document driven 'spec'. The canonical implementation of Org mode is org-element.el which links to 'see more details about Org' (https://github.com/emacs-mirror/emacs/blob/fe15532eef5e7375fd2d91b92e173e5ed12534ef/lisp/org/org-element.el#L25).

It's as always - as long as the documentation is not automatically generated, or even better - the code generated from the documentation, then there's always a good chance of discrepancy. That's why having a BNF based parser is quite superior in the long run: Less chance of ambiguities between spec and implementation(;  

In any case, Emacs doesn't have the regular `CLOCK` functionality for the shorthand syntax. If I update the times and `C-c C-c` it, there's no update in the calculated time. However, it does certainly parse it correctly for the syntax highlighting, at least(;

TL;DR: Parsing more is never wrong, so it's great that org-parser does it! But I don't think it's valid syntax according to what actually works in Emacs.

Reply:
Alright, we will definitely export/render the timestamps `[]--[]` which Org mode understands :)

Reply:
:+1: